### PR TITLE
add latest dydx pyex; update negative balance logic

### PR DESCRIPTION
**Pull request by MikeHathaway**

### CODE REVIEW

File: market_maker_keeper/dydx_market_maker_keeper.py [hunk 1]
```diff
@@ -102,24 +102,28 @@ def our_available_balance(self, our_balances: dict, token: str) -> Wad:
...
+        wei_balance = float(list(filter(lambda x: x['currency'] == token.upper(), our_balances))[0]['wei'])
 
         ## DyDx can have negative balances from native margin trading
+        is_negative = False
+        if wei_balance < 0:
+           is_negative = True
+
+        balance = from_wei(abs(int(wei_balance)), 'ether')
+
         # reconvert Wad to negative value if balance is negative
```
Comment (EdNoepel) [inline market_maker_keeper/dydx_market_maker_keeper.py:114]: Does `Wad` or `from_wei` not support negative numbers?

File: market_maker_keeper/cex_api.py [hunk 2]
```diff
@@ -36,25 +36,28 @@ class CEXKeeperAPI:
     Define a common abstract API for keepers on centralized and hybrid exchanges
     """
 
-    def __init__(self, arguments: Namespace, pyex_api: PyexAPI):
+    def __init__(self, arguments: Namespace, pyex_api: PyexAPI, standard_pyex: bool):
```
Comment (EdNoepel) [inline market_maker_keeper/cex_api.py:39]: The `standard_pyex` variable tastes a little off.  Does it mean `uses_order_history_reporter_and_order_book_manager`?  Are those two facilities inseparable?  Should we have separate methods for `init_history_reporter` and `init_order_book_manager` and let subclasses override behavior where needed?

File: market_maker_keeper/dydx_market_maker_keeper.py (hunk 1 above)
Comment (MikeHathaway) [reply market_maker_keeper/dydx_market_maker_keeper.py:114]: yep, `from_wei` does not support negative numbers

File: market_maker_keeper/cex_api.py (hunk 2 above)
Comment (MikeHathaway) [reply market_maker_keeper/cex_api.py:39]: Good point. Creating an `init_order_book_manager` method that could be subclassed would be cleaner.

Comment (MikeHathaway) [reply market_maker_keeper/cex_api.py]: I continued passing arguments and `pyex_api` for the typechecking in init, even though they weren't strictly necessary

**Pull request merged**